lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -479,8 +479,11 @@ listTransactions w (ApiT wid) mStart mEnd mOrder = do
     txs <- liftHandler $ W.listTransactions w wid
         (getIso8601Time <$> mStart)
         (getIso8601Time <$> mEnd)
-        (getApiT <$> mOrder)
+        (maybe defaultSortOrder getApiT mOrder)
     return $ map mkApiTransactionFromInfo txs
+  where
+    defaultSortOrder :: SortOrder
+    defaultSortOrder = Descending
```

Comment:
The notion of a default sort order for `listTransactions` now only lives here in the Api. `WalletLayer.listTransactions` doesn't have a default sort-order anymore.